### ISSUE TITLE
libarchive fails to configure on Mavericks

### ISSUE THREAD

**Issue by jwiegley** [NixOS/nixpkgs#5608]
When trying to build `libarchive` on Mavericks, I'm seeing the following configuration failure:

```
checking size of wchar_t... configure: error: in `/private/var/folders/sq/906k53jd3w7cd5rf9fpc0g_00000gn/T/nix-build-libarchive-3.1.2.drv-0/libarchive-3.1.2':
configure: error: cannot compute sizeof (wchar_t)
```

**Comment by jwiegley** [comment]
This is actually the exact same `zlib` failure as before, so I think the rname needs to be correctly set in the library, rather than wrapping as I did for `curl`:

```
configure:15752: checking size of wchar_t
configure:15757: cc -o conftest -g -O2  -I/nix/store/75k2c0qdnsnwbs59mksb1riwk22zwqav-libxml2-2.9.2/
include/libxml2  conftest.c -llzma -lcharset -lbz2 -lz  -L/nix/store/75k2c0qdnsnwbs59mksb1riwk22zwqa
v-libxml2-2.9.2/lib -lxml2 -lz -lpthread -liconv -lm >&5
configure:15757: $? = 0
configure:15757: ./conftest
dyld: Library not loaded: libz.so.1.2.8
  Referenced from: /private/var/folders/sq/906k53jd3w7cd5rf9fpc0g_00000gn/T/nix-build-libarchive-3.1
.2.drv-0/libarchive-3.1.2/./conftest
  Reason: image not found
./configure: line 1767: 73094 Trace/BPT trap: 5       ./conftest$ac_exeext
configure:15757: $? = 133
configure: program exited with status 133
```

**Comment by jwiegley** [comment]
These two issues are properly fixed in f8bd5bb.
